File: backend/src/main/java/zw/co/hisolutions/core/service/SkillService.java
package zw.co.hisolutions.core.service;
 
import zw.co.hisolutions.core.entity.Skill;
import zw.co.hisolutions.common.service.GenericService;

public interface SkillService extends GenericService<Skill, Long>{
    public Skill findByName(String name);   
}

File: frontend/src/app/shared/services/rest/skill.service.ts
import { Injectable } from '@angular/core'; 
import {HttpClient} from '@angular/common/http';
import {DataService} from 'shared/services/data.service';

@Injectable()
export class SkillService extends DataService {
    
    constructor(httpClient: HttpClient) {
        const resourcePath =  '/rest/skills'; 
        super(httpClient, resourcePath);
    }

}
